open_optical_gating/cli/file_optical_gater.py

```diff
@@ -44,22 +45,29 @@ def load_data(self, filename):
         self.data = io.imread(filename)
         self.height, self.width = self.data[0].shape
         self.framerate = self.settings["brightfield_framerate"]
+        self.start_time = time.time()
 
-    def next_frame(self, force_framerate=True):
+    def next_frame(self, force_framerate=False):
         """This function gets the next frame from the data source, which can be passed to analyze()"""
         # Force framerate to match the brightfield_framerate in the settings
         # This gives accurate timings and plots
-        # TODO when timestamp moves into a frame object this can probably be defaulted to False
-        if force_framerate and (len(self.timestamp) > 0):  # only do once we have frames
-            while (time.time() - self.initial_process_time_s - self.timestamp[-1]) < (
-                1 / self.settings["brightfield_framerate"]
-            ):
+        if force_framerate and (
+            len(self.frame_history) > 0
+        ):  # only do once we have frames
+            while (
+                time.time()
+                - self.start_time
+                - self.frame_history[-1].metadata["timestamp"]
+            ) < (1 / self.settings["brightfield_framerate"]):
                 time.sleep(1e-5)

```

Comment:
Should the 1e-5 be here or should there be variable instead which is determined in a settings file or elsewhere in the code? I can't imagine people needing to change this though...

Reply:
Nah, it's just "any small time" so that it doesn't do a million loops a second... only 10000 lol... should probably do this without a loop at all now I'm thinking about it... one sec....